### DISCUSSION TITLE
i219v-2 network failure

### DISCUSSION THREAD

**Discussion by iggyvolz** [ipxe/ipxe#458] [Q&A]
Apologies if this is just me doing something wrong (I'm not terribly familiar with the concept of network/PXE booting) - however I have not been able to get iPXE to have any sort of network connection on an i219v-2, and looking from https://forum.ipxe.org/showthread.php?tid=14972 https://forum.ipxe.org/showthread.php?tid=7804 https://lists.ipxe.org/pipermail/ipxe-devel/2019-February/006570.html and https://github.com/ipxe/ipxe/commit/546dd51de8459d4d09958891f426fa2c73ff090d, this looks to be a recurring issue with no clear workaround or resolution.

Loading the iPXE LiveUSB and running `ifopen net0` returns (with several debug modes on):
```
INTEL 0xec49c WARNING: applying reset hang workaround
INTEL 0xec49c ring 03800 is at [c6e68600,c6e68700)
INTEL 0xec49c ring 02800 is at [c6e68700,c6e68800)
INTEL 0xec49c RX 0 is [c6e68800,c6e69000)
INTEL 0xec49c RX 1 is [c6e69000,c6e69800)
INTEL 0xec49c RX 2 is [c6e6a000,c6e6a800)
INTEL 0xec49c RX 3 is [c6e6a800,c6e6b000)
INTEL 0xec49c RX 4 is [c6e6b000,c6e6b800)
INTEL 0xec49c RX 5 is [c6e6b800,c6e6c000)
INTEL 0xec49c RX 6 is [c6e6c000,c6e6c800)
INTEL 0xec49c RX 7 is [c6e6c800,c6e6d000)
INTEL 0xec49c link status is 40080680
INTEL 0xec49c unexpected ICR 00000102
iPXE > 
```

An ifstat returns:
```
net0: <<my mac address>> using i219v-2 on 0000:00:1f.6 (open)
  [Link:down, TX:0 TXE:1 RX:0 RXE:0]
  [Link status: Down (http://ipxe.org/38086101)]
  [TXE: 1 x "Operation not supported (http://ipxe.org/3c654003)"]
```

dhcp simply returns:
```
Waiting for link-up on net0.................. Down (http://ipxe.org/38086101)
INTEL 0xec49c forced MAC reset (00180260/40080680 was 00180240/40080680)
```

I am able to get a DHCP connection booting into both Linux and Windows on this computer automatically, so the problem appears to be in iPXE.

Potentially of note is that the machine uses Legacy Boot (our existing network/PXE server has issues network booting over DHCP with UEFI however we haven't been able to diagnose the issue - hence looking into iPXE).

Any help/ideas/potential fixes would be welcome - I don't have much experience with any sort of network drivers but I could try!

**Comment by NiKiZe** [comment]
What is the link speed, and have you tried to force a different speed on the interface?

**Reply by iggyvolz** [reply]
I'm not sure how to do that or where to find that - not seeing anything relevant in the config and unfortunately I don't have control over the network.

**Reply by NiKiZe** [reply]
Check the switch, get a temporary switch to but in between that you do control.
